File: app/src/main/java/ru/vincetti/vimoney/data/sqlite/CurrentDao.java
package ru.vincetti.vimoney.data.sqlite;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ru.vincetti.vimoney.data.models.CurrencyModel;

@Dao
public interface CurrentDao {
    @Query("Select * from currency ORDER BY id ASC")
    LiveData<List<CurrencyModel>> loadAllCurrency();

    @Query("Select * from currency WHERE code = :code LIMIT 1")
    LiveData<CurrencyModel> loadCurrencyByCode(int code);

    @Insert
    void insertCurrency(CurrencyModel cur);
}

File: app/src/main/java/ru/vincetti/vimoney/data/sqlite/CurrentDao.kt
package ru.vincetti.vimoney.data.sqlite

import androidx.room.*
import ru.vincetti.vimoney.data.models.CurrencyModel

@Dao
interface CurrentDao {

    @Query("Select * from currency ORDER BY id ASC")
    suspend fun loadAllCurrency(): List<CurrencyModel>?

    @Query("Select * from currency WHERE code = :code LIMIT 1")
    suspend fun loadCurrencyByCode(code: Int): CurrencyModel?

    @Insert
    suspend fun insertCurrency(cur: CurrencyModel)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateCurrency(cur: CurrencyModel)

    @Delete
    suspend fun deleteCurrency(cur: CurrencyModel)
}
